keps/79-hierarchical-cohorts/README.md

```diff
@@ -0,0 +1,246 @@
+# KEP-79: Hierarchical Cohorts
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Stories (Optional)](#user-stories-optional)
+    - [Story 1](#story-1)
+    - [Story 2](#story-2)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [Test Plan](#test-plan)
+      - [Prerequisite testing updates](#prerequisite-testing-updates)
+    - [Unit Tests](#unit-tests)
+  - [Graduation Criteria](#graduation-criteria)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+## Summary
+
+Introduce Cohort top-level object to allow setting multi-level quota 
+hierarchy, with advanced borrowing, lending mechanisms.
+
+## Motivation
+
+The current 2-level hierarchy (ClusterQueues and Cohorts) is not expressive
+enough to handle complex use cases of large organizations with tree-like
+team and quota/budget structures.
+
+### Goals
+
+* Create a multi-level hierarchy for advanced quota management. 
+* Be compatible with the existing ClusterQueue API and mechanics.
+* Allow setting constraints about borrowing and lending at all levels.
+* Provide quota for groups of Queues.
```

Comment:
do we have any component to check if the quota tree is valid?

Reply:
What do you mean by that? Any non-negative quota is, in theory, valid...

Reply:
That's an implementation detail that is discussed in the risks section.